compiler/noirc_evaluator/src/ssa/opt/inlining/inline_info.rs

```diff
@@ -163,16 +163,21 @@ fn compute_function_should_be_inlined(
     let function = &ssa.functions[&func_id];
     let assert_constant_id = function.dfg.get_intrinsic(Intrinsic::AssertConstant).copied();
     let static_assert_id = function.dfg.get_intrinsic(Intrinsic::StaticAssert).copied();
-    let contains_static_assertion = function.reachable_blocks().iter().any(|block| {
-        function.dfg[*block].instructions().iter().any(|instruction| {
-            match &function.dfg[*instruction] {
-                Instruction::Call { func, .. } => {
-                    Some(*func) == assert_constant_id || Some(*func) == static_assert_id
+
+    let contains_static_assertion = if assert_constant_id.is_none() && static_assert_id.is_none() {
+        false
+    } else {
+        function.reachable_blocks().iter().any(|block| {
```

Comment:
Can we not get rid of this logic entirely? Because if there are IDs for either intrinsic then we do contain a static assertion?

Reply:
I thought the same. But I was wondering if the intrinsic was used in the function, then a simplification removed that call, I think the intrinsic would still be registered but unused, right?

Reply:
Yes, good call. We never mutate the `intrinsics` map`.